### PR TITLE
add capability to use MessageModel as the input of the Azure Service Bus Sender

### PR DESCRIPTION
**Pull request by jdevillard**
This PR give the ability for the user to specify a MessageModel object as the input of the Activity.

This allow the user to customize a lot of property of the Brocker Service Message like SystemProps and User Props.

Today this can be use in Code and we can think about new Activity like CreateServiceBusMessage to have UI to create this kind of object using the Designer

### CODE REVIEW

**Comment by sfmskywalker** [comment]
I'm wondering if it wouldn't be better to simply expose the message properties directly from the UI (under the Advanced tab)? Although it's clever what you did, it kinda mixes the "payload" input (currently named "Message" on the activity) with the metadata.

So perhaps you can simply model the `SendAzureServiceBusQueueMessage` and `SendAzureServiceBusTopicMessage` activities after the Azure Service Bus `Message` model by replicating each property.

Then each property is available from the UI and of course from the fluent API.

And to prevent too much copy & paste of these properties, maybe you can add a base class activity called e.g. `AzureServiceBusActivity`.

Thoughts?

**Comment by jdevillard** [comment]
ok, let's go for this.

Just, about the Key/Value pair in user props , what UI is used for this?

**Comment by jdevillard** [comment]
Base class was added with metadata mapping, SendAzureServiceBusQueueMessage and SendAzureServiceBusTopicMessage was simplified.

File: src/activities/Elsa.Activities.AzureServiceBus/Activities/SendAzureServiceBusMessage/AzureServiceBusSendActivity.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        [ActivityInput(SupportedSyntaxes = new[] { SyntaxNames.JavaScript, SyntaxNames.Liquid })] 
+        public DateTime ExpiresAtUtc { get; set; }
+        [ActivityInput(SupportedSyntaxes = new[] { SyntaxNames.JavaScript, SyntaxNames.Liquid })] 
+        public TimeSpan TimeToLive { get; set; }
+        [ActivityInput(SupportedSyntaxes = new[] { SyntaxNames.JavaScript, SyntaxNames.Liquid })] 
+        public string? ReplyToSessionId { get; set; }
+        [ActivityInput(SupportedSyntaxes = new[] { SyntaxNames.JavaScript, SyntaxNames.Liquid })] 
+        public DateTime ScheduledEnqueueTimeUtc { get; set; }
+        [ActivityInput(DefaultSyntax = SyntaxNames.Json, SupportedSyntaxes = new[] { SyntaxNames.JavaScript, SyntaxNames.Liquid,SyntaxNames.Json },UIHint = ActivityInputUIHints.MultiLine)] 
+        public IDictionary<string,Object> UserProperties { get; set; } = new Dictionary<string, object>();
```
Comment (sfmskywalker) [inline src/activities/Elsa.Activities.AzureServiceBus/Activities/SendAzureServiceBusMessage/AzureServiceBusSendActivity.cs:58]: Should we move these to the Advanced category?

**Review by sfmskywalker — APPROVED** [review]

File: src/activities/Elsa.Activities.AzureServiceBus/Activities/SendAzureServiceBusMessage/AzureServiceBusSendActivity.cs (hunk 1 above)
Comment (jdevillard) [reply src/activities/Elsa.Activities.AzureServiceBus/Activities/SendAzureServiceBusMessage/AzureServiceBusSendActivity.cs]: yep 👍 

done!

**Comment by sfmskywalker** [comment]
> Just, about the Key/Value pair in user props , what UI is used for this?

We don't have an appropriate UI for that, but what will work for the time being is using the `ActivityInputUIHints.MultiLine` UI with a `DefaultSyntax = SyntaxNames.Json` combination (and `SupportedSyntaxes = new[] { SyntaxNames.Json, SyntaxNames.JavaScript, SyntaxNames.Liquid }`) - exactly like what we do for the `RequestHeaders` property on `SendHttpRequest`.

The user can then enter key/value pairs with JSON syntax like this:

```json
{
   "key1": "value1",
   "key2": "value2",
   //...etc.
}
```

**Pull request merged**